pkg/apis/configuration/validation/validation_test.go

```diff
@@ -1658,16 +1689,61 @@ func TestValidateTime(t *testing.T) {
 	}
 }
 
+func TestValidateOffset(t *testing.T) {
+	var validInput = []string{"", "1", "10k", "11m", "1K", "100M", "5G"}
+	for _, test := range validInput {
+		allErrs := validateOffset(test, field.NewPath("offset-field"))
+		if len(allErrs) != 0 {
+			t.Errorf("validateOffset(%q) returned an error for valid input", test)
+		}
+	}
+
+	var invalidInput = []string{"55mm", "2mG", "6kb", "-5k", "1L", "5Gb"}
+	for _, test := range invalidInput {
+		allErrs := validateOffset(test, field.NewPath("offset-field"))
+		if len(allErrs) == 0 {
+			t.Errorf("validateOffset(%q) didn't return error for invalid input.", test)
+		}
+	}
+}
+
+func TestValidateBuffer(t *testing.T) {
+	validbuff := &v1alpha1.UpstreamBuffers{Number: 8, Size: "8k"}
```

Comment:
Here and in `TestValidateOffset` I think creating a for loop with all the cases will reduce a lot of duplicated code. For example, have a look at `TestValidateSize` you could do something like:

```golang
	var validBuffers = []v1alpha1.UpstreamBuffers{
		v1alpha1.UpstreamBuffers{
			Number: 8,
			Size:   "8k",
		},
		// etc
	}
```

And then run over the cases in a for loop like the other test cases

Also, normally in the rest of this file we separated valid and invalid test cases. For instance: `TestValidateBuffer` will have all the valid cases (if is only 1 we don't need a for loop) and `TestValidateBufferInvalid` will have the invalid cases. Same applies for `TestValidateSize` I think for consistency we can do the same here. Let me know if that makes sense.